src/trace/entry.zig

```diff
@@ -1,217 +1,227 @@
-const testing = std.testing;
 const AtomicBool = std.atomic.Value(bool);
+const Fields = std.ArrayList(u8);
 
 pub const Entry = union(enum) {
-    standard: *StandardEntry,
+    standard: *StdEntry,
+    testing: *TestingEntry,
     noop,
 
     const Self = @This();
 
-    pub fn init(allocator: std.mem.Allocator, channel: *Channel(*StandardEntry), max_level: Level) Self {
-        return .{ .standard = StandardEntry.init(allocator, channel, max_level) };
+    pub fn init(allocator: std.mem.Allocator, channel: *Channel(logfmt.LogMsg), log_level: Level) Self {
+        return .{ .standard = StdEntry.init(allocator, channel, log_level) };
     }
 
     pub fn deinit(self: Self) void {
         switch (self) {
-            .standard => |entry| {
-                entry.deinit();
-            },
             .noop => {},
+            .standard => |impl| impl.deinit(),
+            .testing => |impl| impl.deinit(),
         }
     }
 
     pub fn field(self: Self, name: []const u8, value: anytype) Self {
         switch (self) {
+            .noop => {
+                return self;
+            },
             .standard => |entry| {
                 _ = entry.field(name, value);
                 return self;
             },
-            .noop => {
+            .testing => |entry| {
+                _ = entry.field(name, value);
                 return self;
             },
         }
     }
 
-    pub fn debugf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        self.logf(.debug, fmt, args);
-    }
-
-    pub fn errf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        self.logf(.err, fmt, args);
+    pub fn log(self: Self, msg: []const u8) void {
+        switch (self) {
+            .noop => {},
+            .standard => |impl| impl.log(msg),
+            .testing => |impl| impl.log(msg),
+        }
     }
 
-    pub fn warnf(self: Self, comptime fmt: []const u8, args: anytype) void {
-        self.logf(.warn, fmt, args);
+    pub fn logf(self: Self, comptime fmt: []const u8, args: anytype) void {
+        switch (self) {
+            .noop => {},
+            .standard => |impl| impl.logf(fmt, args),
+            .testing => |impl| impl.logf(fmt, args),
+        }
     }
+};
 
-    pub fn infof(self: Self, comptime fmt: []const u8, args: anytype) void {
-        self.logf(.info, fmt, args);
-    }
+pub const StdEntry = struct {
+    allocator: std.mem.Allocator,
+    scope: ?[]const u8,
+    log_level: Level,
+    fields: Fields,
+    exit_sig: std.atomic.Value(bool),
+    channel: *Channel(logfmt.LogMsg),
+    const Self = @This();
 
-    pub fn info(self: Self, msg: []const u8) void {
-        self.log(.info, msg);
+    pub fn init(
+        allocator: std.mem.Allocator,
+        scope: ?[]const u8,
+        channel: *Channel(logfmt.LogMsg),
+        log_level: Level,
+    ) *Self {
+        const self = allocator.create(Self) catch @panic("could not allocate.Create Entry");
```

Comment:
It may be preferable to avoid an allocation here and pass by value instead.

Reply:
Done in [df7c40b](https://github.com/Syndica/sig/pull/277/commits/df7c40ba70b3dd51da55bc7b535a0d84ef3a1817)